src/server/readpool/config.rs

```diff
@@ -48,114 +32,22 @@ pub struct Config {
     pub stack_size: ReadableSize,
 }
 
-impl Default for Config {
-    fn default() -> Config {
-        Config {
-            high_concurrency: DEFAULT_CONCURRENCY,
-            normal_concurrency: DEFAULT_CONCURRENCY,
-            low_concurrency: DEFAULT_CONCURRENCY,
-            max_tasks_high: DEFAULT_MAX_TASKS_PER_CORE * DEFAULT_CONCURRENCY,
-            max_tasks_normal: DEFAULT_MAX_TASKS_PER_CORE * DEFAULT_CONCURRENCY,
-            max_tasks_low: DEFAULT_MAX_TASKS_PER_CORE * DEFAULT_CONCURRENCY,
-            stack_size: ReadableSize::mb(DEFAULT_STACK_SIZE_MB),
-        }
-    }
-}
-
 impl Config {
-    pub fn default_for_coprocessor() -> Config {
-        let cpu_num = sys_info::cpu_num().unwrap();
-        let concurrency = if cpu_num > 8 {
-            (f64::from(cpu_num) * 0.8) as usize
-        } else {
-            DEFAULT_COP_CONCURRENCY
-        };
-        Config {
+    /// Only used in tests.
+    pub fn default_with_concurrency(concurrency: usize) -> Self {
```

Comment:
It specifies a default value (stack_size) so that I think it is not purely "with_concurrency". Also if we have more config fields, they will all contains a default value.